### ISSUE TITLE
feature: syncer able to create self-sufficient targets

### ISSUE THREAD

**Issue by MikeSpreitzer** [kcp-dev/kcp#2682]
### Feature Description

Background: in edge-mc, we want each edge cluster to be able to operate independently of the center and any service providers.  This is so the edge cluster can tolerate long periods of disconnection and support data sovereignty requirements.

The feature request here is to generalize the syncer so that it has the option for the objects it creates in the pcluster to not be connected to the apiserver back in the source workspace but rather be connected to the apiserver in the pcluster.

### Proposed Solution

Extend the syncer's configuration with a boolean option regarding this behavior.

### Alternative Solutions

- Make the option more fine-grained, perhaps on an object-by-object basis or resource-by-resource basis.
- Let the boolean option appear on the source object, namespace, or workspace.
- Have edge-mc copy and modify the syncer source.
- Factor the syncer source so that TMC and EMC can share most of it while not requiring either syncer binary to support both behaviors.

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

https://github.com/kcp-dev/edge-mc/pull/148

**Comment by ncdc** [comment]
For clarity: the ask is for a workload (e.g. `Deployment`) to optionally **not** go through the deployment mutator to change `KUBERNETES_SERVICE_HOST` and other relevant settings. In other words, the deployment synced to the workload cluster needs to talk to the workload cluster, not back to kcp.

**Comment by MikeSpreitzer** [comment]
Yes, and it is not just about Deployment.  It is about anything that leads to Pods in the workload cluster.

**Comment by MikeSpreitzer** [comment]
When discussed at the Jan 31 community meeting, it was agreed to pursue the following alternate approach.  Let TMC and EMC each maintain their own syncer binary that behaves in the way appropriate for each, while sharing as much code as possible (which is expected to be very nearly all at first, probably diverging more as time goes on).

**Comment by MikeSpreitzer** [comment]
On second thought, making self-sufficient edge clusters raises deeper issues --- briefly outlined at https://github.com/kcp-dev/edge-mc/pull/148#issuecomment-1424734973

**Comment by ezrasilvera** [comment]
@MikeSpreitzer I'm not sure what's the bottom line here - should we proceed and implement this and deal with the necessary RBAC and permission support needed on the pCluster later? Or should we pause with this? 
I think we should proceed (and wanted to start working on that), what do you think ?
